### PR TITLE
Update MLflow UI

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Update MLflow UI.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Pinning feature:

https://user-images.githubusercontent.com/17039389/198252650-45358057-3bbc-4658-9488-9a945dd39647.mov

Other pages/components also look ok:

https://user-images.githubusercontent.com/17039389/198264902-dbd24b86-6512-4c27-b019-cfa066fb1972.mov

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/search_utils.py [hunk 1]
```diff
@@ -211,7 +211,12 @@ def _valid_entity_type(cls, entity_type):
     @classmethod
     def _get_identifier(cls, identifier, valid_attributes):
         try:
-            entity_type, key = identifier.split(".", 1)
+            tokens = identifier.split(".", 1)
+            if len(tokens) == 1:
+                key = tokens[0]
+                entity_type = cls._ATTRIBUTE_IDENTIFIER
+            else:
+                entity_type, key = tokens
```
Comment (harupy) [inline mlflow/utils/search_utils.py:219]: @dbczumar This is a change to support a key-only filter string such as `run_id IN (...)`.

File: mlflow/server/js/craco.config.js [hunk 2]
```diff
@@ -194,7 +190,19 @@ module.exports = function ({ env }) {
         ],
       ],
     },
-    ...(proxyTarget && {
+    ...(process.env.MLFLOW_DEV_PROXY_MODE && {
```
Comment (hubertzub-db) [inline mlflow/server/js/craco.config.js:193]: This code block is not required here but shouldn't do any harm. We will OSS-guard it on next export.

**Review by hubertzub-db — APPROVED** [review]
Looks lgtm overall, there are minor pieces we could EDGE-guard but it's nothing critical nor secretive so we can fix them next time

File: mlflow/server/js/craco.config.js (hunk 2 above)
Comment (harupy) [reply mlflow/server/js/craco.config.js:193]: Makes sesnse, thanks for the reivew!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Comment by dbczumar** [comment]
> Question re: pinning -- it looks like nothing was changed server-side to support this.
> 
> Since MLflow doesn't have users, I assume it was done this way so that the state of pinned runs is only applied to a given browser? (i.e. so that various users of the system don't fight over what runs are pinned within an experiment?)

@ddl-ebrown Thank you for reaching out! That's correct. MLflow doesn't yet have the concept of users, so pinning is implemented in browser local storage.

**Pull request merged**